scrypto/src/component/stubs.rs

```diff
@@ -150,51 +151,55 @@ extern_blueprint_internal! {
             owner_role: OwnerRole,
             address_reservation: Option<GlobalAddressReservation>,
         ) -> Global<Account>;
-        fn create() -> (Global<Account>, Bucket);
+        fn create() -> (Global<Account>, NonFungibleBucket);
     },
     {
-        fn securify(&mut self) -> Bucket;
+        fn securify(&mut self) -> NonFungibleBucket;
         fn lock_fee(&mut self, amount: Decimal);
         fn lock_contingent_fee(&mut self, amount: Decimal);
         fn deposit(&mut self, bucket: Bucket);
         fn deposit_batch(&mut self, buckets: Vec<Bucket>);
-        fn withdraw(&mut self, resource_address: ResourceAddress, amount: Decimal) -> Bucket;
+        fn withdraw(&mut self, resource_address: ResourceManager, amount: Decimal) -> Bucket;
```

Comment:
I was a bit shocked by the change, despite knowing that we're introducing ResourceManager, FungibleResourceManager and NonFungibleResourceManager.

Are we feeling comfortable with replacing `resource_address` with resource manager? Curious to learn how other people feel about this.

Reply:
It's interesting. If we're saying that a fungible and non fungible resources are `FungibleResourceManager` and `NonFungibleResourceManager` then it makes sense that an undecided resource would be a `ResourceManager`. But I see your point on this being a bit of a big change.

Reply:
Perhaps this is one where traits can be useful? As an example, with some changes to the `extern_blueprint` macro I imagine that we can support `resource_address: impl Into<ResourceAddress>` which seems like a more correct argument type for this function.

Reply:
I think in my mind there are two different things: addresses (identifiers) and stubs, similar to component address and `Global<T>`.  

The question is more about if we should use address or stubs for the interfaces.

Stubs would make Scrypto-to-Scrypto interaction easier, as there is no more conversion needed and everything is type checked.

On the other side, manifest developers/readers will need to map resource manager stubs to addresses.

Reply:
Agreed to revert these changes (replacing `ResourceAddress` with `ResourceManager`) in this PR.
We will get back to this in separate PR if the team reaches consensus.